internal/client/download.go

```diff
@@ -22,18 +23,28 @@ func Download(version string) (string, error) {
 	downloadURL := fmt.Sprintf("%s/%s/%s", TerraformReleasesUrl, version, fileName)
 
 	var err error
-	for attempt := 1; attempt <= MaxRetries; attempt++ {
+	var attempt int
+	for attempt = 1; attempt <= MaxRetries; attempt++ {
 		if err = downloadWithRetry(downloadURL, downloadLocation, fileName); err == nil {
 			fmt.Printf("Terraform version %s downloaded successfully\n", helpers.ColoredVersion(version))
 			return fmt.Sprintf("%s/%s", downloadLocation, fileName), nil
 		}
-
+		if strings.Contains(err.Error(), "404 Not Found") {
```

Comment:
Since we know that releases before 1.0 (?) do not have a ARM build for Mac, I'd rather see an implementation that specifically checks for this instead of entangling it with the download retry logic.

Reply:
For checking 404, can we use the HTTP response header?

Reply:
Reminded me of https://abenezer.org/blog/hyrum-law-in-golang

Reply:
Apart from asserting an error, if we consider moving forward with this PR and address [this comment](https://github.com/tfversion/tfversion/pull/58#issuecomment-2466700275), I would consider 4XX as non-retryable and 5XX as retryable for the retry logic in this proposal—to avoid asserting single headers or error messages from the response body.